### PR TITLE
Update of Button and Pin input blocks

### PR DESCRIPTION
**Pull request by Amerlander**
- adds the event types usable for Button and Pin events
- merges the `onPinPressed` & the `onPinReleased` blocks into a new `onPinEvent` block.
- creates a new `onButtonEvent` block, which replaces the old `onButtonPressed` one (which actually was click-listener instead of pressed)
- change the game.ts to use the new onButtonEvent function
- the _old_ blocks are left on the code side for compatibility

### CODE REVIEW

**Comment by Amerlander** [comment]
ok, I see. I have to change the tests first.

**Comment by pelikhan** [comment]
I recommend having an enum that hides DAL

**Comment by Amerlander** [comment]
thank you, used the enum now instead of DAL.

I also brought back the deprecated blocks, so they are displayed in block view when old hex files or js code is imported.

**Comment by pelikhan** [comment]
I would recommend to create a new extension for these blocks to simplify your life w.r.t. to merging Microbit changes. Also we are trying to keep all the editors compatible so this should probably go into common packages.

**Comment by Amerlander** [comment]
I'm not sure how this simplifies my life, since I'm still working myself into this and have to figure out how things work in pxt.

- If I create an extension I would still want to disable the existing _button-_ and _pin-pressed-blocks_, since I don't want to have four _same same but different_ blocks. So all the docs and tutorials would still need to change in compare to Microbit.

- common packages could be a way. The `buttons.cpp` also already has the button events as enum, but I don't know yet what to change there and after how to bind it into pxt-calliope. And still, if Microbit doesn't sneak in to this, it wouldn't be easier to merge their changes later at all?

My reason for these changes are, that the block `on Button A pressed` in fact listens to `click` and not `down` event. And instead oft just renaming or changing the event I thought iit would be nice to have a choice. And since I wasn't able to put the event as an optional parameter into the existing blocks I created new ones to replace the existing.

The Microbit pin and button event listeners have the same mix-up of `pressed` and `click`, so I think it would make sense to change it there too.
If it makes sense to put these changes in common packages, could you assist me with getting them in there? And when Microbit makes use of them we can just merge their changes into pxt-calliope which makes sure we're not going apart.

**Comment by MKleinSB** [comment]
I remembered this issue: https://github.com/microsoft/pxt-microbit/issues/963
Doesn´t it affect your problem, too?

**Comment by pelikhan** [comment]
ready?

File: docs/examples/gameofLife.md [hunk 1]
```diff
@@ -22,13 +22,13 @@ Here's a program that simulates cell life in the LED matrix. Use button ``A`` fo
 let lifeChart: Image = null
 
 //Use button A for the next iteration of game of life
-input.onButtonPressed(Button.A, () => {
+input.input.onButtonEvent(Button.A, ButtonEvent.Click, () => {
```
Comment (pelikhan) [inline docs/examples/gameofLife.md]: double input?

**Comment by pelikhan** [comment]
This change will make future merging from upstream microbit harder.

File: docs/examples/gameofLife.md (hunk 1 above)
Comment (Amerlander) [reply docs/examples/gameofLife.md]: no, the other one is the outdated file, so this is what the kids will deal with.

File: docs/examples/gameofLife.md [hunk 2]
```diff
@@ -22,13 +22,13 @@ Here's a program that simulates cell life in the LED matrix. Use button ``A`` fo
 let lifeChart: Image = null
 
 //Use button A for the next iteration of game of life
-input.onButtonPressed(Button.A, () => {
+input.input.onButtonEvent(Button.A, DAL.MICROBIT_BUTTON_EVT_CLICK, () => {
```
[the remark this answers is not in the document]
Comment (Amerlander) [reply docs/examples/gameofLife.md]: yes, this is outdated and changed within https://github.com/microsoft/pxt-calliope/pull/89/commits/07fe2645cf07f42d8242877163ad4c1f99e00768

File: docs/examples/gameofLife.md (hunk 1 above)
Comment (pelikhan) [reply docs/examples/gameofLife.md]: They have to type Input twice?

Comment (Amerlander) [reply docs/examples/gameofLife.md]: ah, sorry. Yes, that's wrong. I`ll change that.

**Comment by MKleinSB** [comment]
@Amerlander I agree to pelikhan. Wait with the update of the pinevent. The standard Calliope user doesn´t see the difference. It it necessary to keep merging of microbit source easy. The Pinevents will be (perhaps) updated.  Look, there is a new issue
https://github.com/microsoft/pxt-microbit/issues/2667

**Comment by Amerlander** [comment]
@MKleinSB, of course it would make sense to wait when micro bit is changing there something at all. I just wonder whether there is any plan to do this and how I could help with that.

The Issue on the micro bit side was labeled with "next-release", but then removed. They said in 2018 they will introduce a fix next year, but since then there is not really any progress in the issue and also not an answer to my question. I would be glad if the issue can be closed on the side of micro bit, and we can just check in.

The issue https://github.com/microsoft/pxt-microbit/issues/2667 doesn't seem to target this, since its more about when a click event is triggered and when it is a long click and when its not a click at all (a down event where another event comes in between before the up event is not triggered as click anymore). And there the behavior seems to differ from simulator to hardware.

**Comment by MKleinSB** [comment]
@Amerlander If **you stay** working on makecode.calliope it´s ok ( for me ;-) )

**Comment by pelikhan** [comment]
I would recommend to create a new extension that produces the better blocks then merging would be easier. We should setup a call to discuss these things.

**Comment by Amerlander** [comment]
@MKleinSB I'm pretty sure I'll stay, even when this isn't the only thing I'm working on.

@pelikhan having a call to discuss this sounds great.

**Pull request closed without merging**